### PR TITLE
Improve navigation, get coordinates if empty on the form

### PR DESCRIPTION
**Pull request by ramirobg94**

### CODE REVIEW

File: src/form/Attributes/Coordinate/Form/index.js [hunk 1]
```diff
@@ -149,6 +149,7 @@ const Form = ({nodeDef}) => {
         handleSaveLocation={handleSaveLocation}
         selectedSrs={selectedSrs}
         surveySrsIndex={surveySrsIndex}
+        hasCurrentLocation={String(newValue?.x || node?.value?.x || '') !== ''}
```
Comment (SteRiccio) [inline src/form/Attributes/Coordinate/Form/index.js:152]: what if newValue?.x is 0? String(newValue?.x || node?.value?.x || '') will be '' and hasCurrentLocation will be false

File: src/screens/Form/components/EntityPage/components/common/EntityPanel/Navigation/component.js [hunk 2]
```diff
@@ -313,28 +353,48 @@ const useGetNext = ({parent}) => {
...
+            parentNode?.meta?.childApplicability || {},
+          ).includes(childIndex);
+        }),
       )
     ) {
-      return survey.nodeDefs[childrenIndex[0]];
+      const validChild = childrenIndex?.find(childIndex => {
+        return !Object.keys(
+          parentNode?.meta?.childApplicability || {},
+        ).includes(childIndex);
```
Comment (SteRiccio) [inline src/screens/Form/components/EntityPage/components/common/EntityPanel/Navigation/component.js]: you can use arena-core Nodes.isChildApplicable(parentNode, childIndex)

File: src/state/app/sagas/index.js [hunk 3]
```diff
@@ -28,14 +28,15 @@ import appSelectors from '../selectors';
...
+  const emailRegex =
+    /[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/;
 
   if (!emailRegex.test(username)) {
     throw Error('Please enter a valid email address as username.');
   }
 
   // if email ends with @gmail.co or @gmail.con ...
-  if (username.endsWith('.co') || username.endsWith('.con')) {
+  if (username.endsWith('gmail.co') || username.endsWith('.con')) {
```
Comment (SteRiccio) [inline src/state/app/sagas/index.js]: why .con? do you mean .com?

**Review by SteRiccio — CHANGES_REQUESTED** [review]

File: src/state/app/sagas/index.js [hunk 4]
```diff
@@ -28,14 +28,15 @@ import appSelectors from '../selectors';
 const DEFAULT_TIME_OUT = 25000;
 
 const validateUsername = username => {
-  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+  const emailRegex =
+    /[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/;
 
   if (!emailRegex.test(username)) {
     throw Error('Please enter a valid email address as username.');
```
Comment (SteRiccio) [inline src/state/app/sagas/index.js:35]: should you use internationalization for this error message?

Comment (ramirobg94) [reply src/state/app/sagas/index.js:35]: this error is internal

File: src/state/app/sagas/index.js (hunk 3 above)
Comment (ramirobg94) [reply src/state/app/sagas/index.js]: remove

**Review by SteRiccio — APPROVED** [review]

**Pull request merged**